src/Classes/Utils/ColorUtils.ts

```diff
@@ -0,0 +1,199 @@
+        if (input.startsWith('rgb')) {
+            const regex = /rgba?\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? {
+                a: parseInt(result[4]),
+                b: parseInt(result[3]),
+                g: parseInt(result[2]),
+                r: parseInt(result[1]),
+            } : undefined;
+        } else if (input.startsWith('hsl')) {
+            const regex = /hsla?\(\s*(\d+)\s*,\s*(\d+)\%\s*,\s*(\d+)\%\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? ColorUtils.hslaToRgba({
+                a: parseInt(result[4]),
+                l: parseInt(result[3]),
+                s: parseInt(result[2]),
+                h: parseInt(result[1]),
+            }) : undefined;
+        }
+        else {
+            // By Tim Down - http://stackoverflow.com/a/5624139/3493650
+            // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+            const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])([a-f\d])?$/i;
+            const hexFixed = input.replace(shorthandRegex, (_m, r, g, b, a) => {
+                return r + r + g + g + b + b + (a ? a + a : "");
+            });
+            const regex = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})?$/i;
+            const result = regex.exec(hexFixed);
+
+            return result ? {
+                a: parseInt(result[4], 16) / 0xFF,
+                b: parseInt(result[3], 16),
+                g: parseInt(result[2], 16),
+                r: parseInt(result[1], 16),
+            } : undefined;
+        }
+    }
+
+    /**
+     * Converts a Hue Saturation Lightness ([[IHsl]]) object in a [[IRgb]] object
+     * @param hsl
+     */
+    public static hslToRgb(hsl: IHsl): IRgb {
+        const result: IRgb = { b: 0, g: 0, r: 0 };
+
+        if (hsl.s === 0) {
+            result.b = hsl.l; // achromatic
+            result.g = hsl.l;
+            result.r = hsl.l;
+        } else {
+            const q = hsl.l < 0.5 ? hsl.l * (1 + hsl.s) : hsl.l + hsl.s - hsl.l * hsl.s;
+            const p = 2 * hsl.l - q;
+
+            result.r = ColorUtils.hue2rgb(p, q, hsl.h + 1 / 3);
+            result.g = ColorUtils.hue2rgb(p, q, hsl.h);
+            result.b = ColorUtils.hue2rgb(p, q, hsl.h - 1 / 3);
+        }
+
+        result.r = Math.round(result.r * 255);
+        result.g = Math.round(result.g * 255);
+        result.b = Math.round(result.b * 255);
+
+        return result;
+    }
+
+    public static hslaToRgba(hsla: IHsla): IRgba {
+        const rgbResult = ColorUtils.hslToRgb(hsla);
+
+        return {
+            a: hsla.a,
+            b: rgbResult.b,
+            g: rgbResult.g,
+            r: rgbResult.r
+        }
+    }
+
+    /**
+     * Generate a random Rgb color
+     * @param min a minimum seed value for all 3 values
+     */
+    public static getRandomRgbColor(min?: number): IRgb {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of public static method not allowed after declaration of private static method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5216054)